include/arch/arm64/cortex_r/arm_mpu.h

```diff
@@ -151,15 +151,15 @@
 #define REGION_RAM_TEXT_ATTR				   \
 	{						   \
 		/* AP, XN, SH */			   \
-		.rbar = P_RO_U_NA_Msk | NON_SHAREABLE_Msk, \
+		.rbar = P_RO_U_RO_Msk | NON_SHAREABLE_Msk, \
```

Comment:
Is there a particular reason you want the RAM region to be visible to everyone? The design of userspace is for `.data` section to be only visible to supervisor threads. User threads must be explicitly allowed to access data via memory domains. This is a security consideration to avoid user threads to read arbitrary data, especially those that must not be exposed.

Reply:
Sorry for the unclear name, the `REGION_RAM_TEXT_ATTR` is for the `.text` section. I am not sure, but this config exactly follows the MMU https://github.com/zephyrproject-rtos/zephyr/blob/79952ad774caf380fca4d24cd12f397b9c337d29/arch/arm64/core/mmu.c#L650 
And so does the following RO section (P_RO_U_RO) https://github.com/zephyrproject-rtos/zephyr/blob/79952ad774caf380fca4d24cd12f397b9c337d29/arch/arm64/core/mmu.c#L656